src/utils/hooks/useRouter.tsx

```diff
@@ -34,6 +35,17 @@ interface RouterResult {
   location: any
 }
 
+const replaceHifenKeys = (parsedQuery: { [key: string]: any }) => {
+  if (Object.keys(parsedQuery).length === 0) return parsedQuery
+  const replacedHifenKeys = Object.keys(parsedQuery).map((key) => {
+    const convertedKey = key.replace(/-([a-z])/g, function (m, w) {
+      return w.toUpperCase()
+    })
+    return { [convertedKey]: parsedQuery[key] }
+  })
+  return replacedHifenKeys.reduce((a, b) => Object.assign({}, a, b))
```

Comment:
Might be nicer to use `Object.fromEntries()` instead of `reduce`?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries

Reply:
Not sure what was the idea, can you try out and show how do you think this can be replace please?

Reply:
`Object.fromEntries()` turns an array of pairs of values (in sub-array) into and key-value pairs in a new object.
The 4 lines above would be:
```
    })
    return [ [convertedKey], parsedQuery[key] ] // Put in array instead of object here
  })
  return Object.fromEntries(replacedKeys)
```

Personally I find it a bit nicer to read, but not a big deal. Up to you.

Reply:
I can change to that, no problem

Reply:
Done (8da61d1)[https://github.com/openmsupply/application-manager-web-app/pull/267/commits/8da61d14f8b8801d1fa511df2f5ae31237db0d0d]